scp.py

```diff
@@ -177,13 +177,21 @@ def putfo(self, fl, remote_path, mode='0644'):
         @type remote_path: str
         @param mode: permissions (posix-style) for the uploaded file
         @type mode: str
+        @param length: length of the file in bytes. If ``None``, the length
+            will be computed using `seek()` and `tell()`.
         """
+        if length is None:
+            pos = fl.tell()
+            fl.seek(0, 2)  # Seek to end
+            length = fl.tell() - pos
+            fl.seek(pos, 0)  # Seek back
```

Comment:
Similarly, ``os.SEEK_SET`` can be used for readability, but this second argument can also be omitted since that's the default.